src/Domain/CoordinateMaps/CoordinateMap.hpp

```diff
@@ -458,8 +467,11 @@ CoordinateMap<SourceFrame, TargetFrame, Maps...>::inv_jacobian_impl(
             });
 
         if (LIKELY(count != 0)) {
-          mapped_point =
-              std::get<(count != 0 ? count - 1 : 0)>(maps)(mapped_point);
+          const auto& map_in_loop =
+              std::get<(count != 0 ? count - 1 : 0)>(maps);
+          if (not map_in_loop.is_identity()) {
+            mapped_point = map_in_loop(mapped_point);
+          }
```

Comment:
Skip the matrix multiplication by the identity matrix for identity maps.  (And again for the Jacobian code below.)